doc/source/data/doc_code/tensor.py

```diff
@@ -0,0 +1,432 @@
+# flake8: noqa
+
+# fmt: off
+# __create_range_begin__
+import ray
+
+# Create a Dataset of tensors.
+ds = ray.data.range_tensor(100 * 64 * 64, shape=(64, 64))
+# -> Dataset(
+#       num_blocks=200,
+#       num_rows=409600,
+#       schema={__value__: <ArrowTensorType: shape=(64, 64), dtype=int64>}
+#    )
+
+ds.take(2)
+# -> [array([[0, 0, 0, ..., 0, 0, 0],
+#            [0, 0, 0, ..., 0, 0, 0],
+#            [0, 0, 0, ..., 0, 0, 0],
+#            ...,
+#            [0, 0, 0, ..., 0, 0, 0],
+#            [0, 0, 0, ..., 0, 0, 0],
+#            [0, 0, 0, ..., 0, 0, 0]]),
+#     array([[1, 1, 1, ..., 1, 1, 1],
+#            [1, 1, 1, ..., 1, 1, 1],
+#            [1, 1, 1, ..., 1, 1, 1],
+#            ...,
+#            [1, 1, 1, ..., 1, 1, 1],
+#            [1, 1, 1, ..., 1, 1, 1],
+#            [1, 1, 1, ..., 1, 1, 1]])]
+# __create_range_end__
+
+# __create_pandas_begin__
+import ray
+from ray.data.extensions.tensor_extension import TensorArray
+
+import pandas as pd
+import numpy as np
+
+# Start with a tabular base dataset.
+ds = ray.data.range_table(1000)
+
+# Create a single TensorArray column.
+def single_col_udf(batch: pd.DataFrame) -> pd.DataFrame:
+    bs = len(batch)
+    arr = TensorArray(np.zeros((bs, 128, 128, 3), dtype=np.int64))
+    return pd.DataFrame({"__value__": arr})
+
+ds.map_batches(single_col_udf)
+# -> Dataset(num_blocks=17, num_rows=1000, schema={__value__: TensorDtype})
```

Comment:
@clarkzinzow seems weird this is just TensorDtype and not showing the shape/dtype details of the Tensor like the Arrow version does, is this a bug?

Reply:
This was added/fixed in this PR: https://github.com/ray-project/ray/pull/26684

That PR was reverted, but once [this unreverting PR](https://github.com/ray-project/ray/pull/26924) is merged, the shape/dtype will be included in the `TensorDtype` repr.